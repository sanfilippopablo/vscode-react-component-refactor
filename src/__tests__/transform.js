const { classToFunction, functionToClass } = require("../transform");

describe("transform", () => {
  test("classToFunction", () => {
    expect(
      classToFunction(
        `
        class MyComponent extends Component {
          render () {
            return <div>a</div>
          }
        }
        `.trim()
      )
    ).toMatchSnapshot();
  });

  test("functionToClass", () => {
    expect(
      functionToClass(
        `
        const MyComponent = (theProps) => {
          const { a, b } = theProps;
          <div>{theProps.c}</div>
        };
        `.trim()
      )
    ).toMatchSnapshot();
  });
});
