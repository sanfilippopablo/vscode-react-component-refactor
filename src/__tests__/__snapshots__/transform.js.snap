// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform classToFunction 1`] = `"const MyComponent = () => <div>a</div>;"`;

exports[`transform functionToClass 1`] = `
"class MyComponent extends React.Component {
  render() {
    const {
      a,
      b
    } = this.props;
    <div>{this.props.c}</div>;
  }

}"
`;
